packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -68,6 +68,26 @@ function nullGuard<T>(fn: () => T): ?T {
   return fn();
 }
 
+const UnsupportedObjectPropertyTypeToInvalidPropertyValueTypeMap = {
+  FunctionTypeAnnotation: 'FunctionTypeAnnotation',
+  VoidTypeAnnotation: 'void',
+  PromiseTypeAnnotation: 'Promise',
+};
+
+function throwUnsupportedObjectPropertyValueTypeAnnotationParserError(
+  moduleName,
+  propertyValue,
+  propertyKey,
+  type,
+) {
+  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
+    moduleName,
+    propertyValue,
+    propertyKey,
+    type,
+  );
+}
+
```

Comment:
Theoreticaly, this code should appear only in a single place. It is exactly the same of the Flow one. 😉